### PR TITLE
Create separate core vs collection contributor guides

### PR DESCRIPTION
**Pull request by samccann**
##### SUMMARY

This is a first pass attempt to separate core vs collection contributor guidelines whilst also creating a general community guide for all.

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs/docsite/rst/ansible_index.rst

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by samccann** [comment]
This is how it looks  now for Ansible (pkg) - 
image

And for ansible-core:
image

**Comment by sivel** [comment]
Just wanted to call out that we've been getting a number of reports that documentation is hard to find when looking for anything ansible-core specific, like roadmaps for example.  Historically because we have only had `/ansible/` for docs, that is where most people still go, and cannot find some of the docs they are searching for.

I feel it probably makes most sense to have ansible-core as the base, and layer on the community stuff.  Otherwise, people will likely only ever rind the contributing docs for ansible-core if someone from core actually hands them the link.

**Comment by samccann** [comment]
Historically, we had two separate sites because core release didn't coincide with Ansible package release. Now that they are more closely aligned, we could reconsider that strategy. The drawback of one site is that we couldn't say update ansible-2.13 to latest until Ansible 6 releases about a month later.

Meanwhile, I can put the core roadmap on the Ansible (/ansible/) docsite as well. And I'll adjust this PR to put the contributing to core guide on the Ansible site as well.  (AKA default to a complete set of docs on /ansible/ and a core only set of docs on /ansible-core/ for now.

File: docs/docsite/rst/community/collection_development_process.rst [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+========================================
+
+If you want to contribute a feature or fix a bug in  a collection, you must open a **pull request** ("PR" for short). GitHub provides a great overview of `how the pull request process works <https://help.github.com/articles/about-pull-requests/>`_ in general. The ultimate goal of any pull request is to get merged and become part of a collection or ``ansible-core``.
+Here's an overview of the PR lifecycle:
+
+* Contributor opens a PR
+* Ansibot reviews the PR
+* Ansibot assigns labels
+* Ansibot pings maintainers
+* Azure Pipelines runs the test suite
```
Comment (marknet15) [inline docs/docsite/rst/community/collection_development_process.rst:38]: Is it worth covering other CI tools used ? Like for some collections where Zuul is used and has some unique labels / comments for triggering retries like the "recheck" comment cmd etc.

Comment (samccann) [reply docs/docsite/rst/community/collection_development_process.rst:38]: in another PR, sure.  At this point, i'm just moving the files around so we have separate guides. Then I'll being  the Great Edit of it all :-)

**Comment by samccann** [comment]
These links may not last but for now - Ansible docs - http://docs.testing.ansible.com/ansible/devel/installation_guide/index.html#
Core docs - http://docs.testing.ansible.com/ansible-core/2.13/community/index.html

File: docs/docsite/rst/community/maintainers.rst [hunk 2]
```diff
@@ -1,8 +1,8 @@
 .. _maintainers:
 
-********************************
-Collection maintainer guidelines
-********************************
+***************************************
+Guidelines for collection naintainers
```
Comment (ariordan-redhat) [inline docs/docsite/rst/community/maintainers.rst]:
```suggestion
Guidelines for collection maintainers
```

**Review by ariordan-redhat — CHANGES_REQUESTED** [review]
Saw one typo; otherwise, LGTM.

**Review by Andersson007 — APPROVED** [review]
LGTM

File: docs/docsite/rst/community/collection_development_process.rst [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+* Developers, maintainers, community review the PR
+* Contributor addresses any feedback from reviewers
+* Developers, maintainers, community re-review
+* PR merged or closed
+
+Automated PR review: ansibullbot
+--------------------------------
+
+Because Ansible receives many pull requests, and because we love automating things, we have automated several steps of the process of reviewing and merging pull requests with a tool called Ansibullbot, or Ansibot for short.
```
Comment (felixfontein) [inline docs/docsite/rst/community/collection_development_process.rst:47]: This only applies to ansible/ansible and very few collection repos. Might be worth mentioning that when this is updated later. (For now - the split up - we can keep it as-is.)

**Review by felixfontein — APPROVED** [review]

**Review by ariordan-redhat — APPROVED** [review]

**Comment by samccann** [comment]
Picked up the other comments here in this quick PR for now so I don't forget entirely - https://github.com/ansible/ansible/pull/76817  and thanks!

**Pull request merged**